pkg/util/deployment/deployment.go

```diff
@@ -61,7 +61,11 @@ func GetOldRCs(deployment extensions.Deployment, c client.Interface) ([]*api.Rep
 		}
 	}
 	requiredRCs := []*api.ReplicationController{}
-	for _, value := range oldRCs {
+	// Note that go reuses the same memory location for every iteration,
+	// which means the 'value' returned from range will have the same address.
+	// Therefore, we should use the returned 'index' instead.
```

Comment:
This is common enough that the comment is probably not necessary. It also doesn't hurt. Up to you!